### PR TITLE
test(recurring-job): snapshot cleanup

### PR DESCRIPTION
**Pull request by c3y1huang**
https://github.com/longhorn/longhorn/issues/5273
https://github.com/longhorn/longhorn/issues/3836

### CODE REVIEW

**Review by yangchiu — APPROVED** [review]
LGTM

**Comment by innobead** [comment]
@Mergifyio backport v1.4.x v1.3.x

File: manager/integration/tests/test_recurring_job.py [hunk 1]
```diff
@@ -1627,6 +1631,264 @@ def test_recurring_job_snapshot(client, batch_v1_api):  # NOQA
...
+@pytest.mark.recurring_job  # NOQA
+def test_recurring_job_snapshot_delete(set_random_backupstore, client, batch_v1_api, volume_name):  # NOQA
+    """
+    Scenario: test recurring job snapshot-delete
+
+    Given volume created, attached, and healthy.
+    And 3 snapshots were created.
+    And volume has 4 snapshots.
+        - 3 user-created
+        - 1 volume-test
```
Comment (innobead) [inline manager/integration/tests/test_recurring_job.py:1643]: It should be `1 volume-head` instad of `volume-test`

**Comment by innobead** [comment]
LGTM. This is great again with the feature complete.

**Pull request merged**